vscode_extension/src/AacTaskProvider.ts

```diff
@@ -0,0 +1,124 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import * as cp from 'child_process';
+import * as vscode from 'vscode';
+import { AacTaskGroup } from './AacTaskGroup';
+
+export class AacTaskProvider implements vscode.TaskProvider {
+    static aacType: string = 'aac';
+    private aacTaskPromise: vscode.ProviderResult<vscode.Task[]> | undefined = undefined;
+
+    constructor() {}
+
+    public provideTasks(): vscode.ProviderResult<vscode.Task[]> {
+        if (!this.aacTaskPromise) {
+            this.aacTaskPromise = getAacTasks();
+        }
+        return this.aacTaskPromise;
+    }
+
+    public resolveTask(_task: vscode.Task): vscode.ProviderResult<vscode.Task> {
+        const task = _task.definition.task;
+        if (task) {
+            const definition: AacTaskDefinition = <any>_task.definition;
+            return new vscode.Task(definition, _task.scope ?? vscode.TaskScope.Workspace, definition.task, 'aac', new vscode.ShellExecution(`aac ${definition.task}`));
+        }
+        return undefined;
+    }
+}
+
+function execShell(command: string, options: cp.ExecOptions): Promise<{ stdout: string; stderr: string }> {
+    return new Promise<{ stdout: string; stderr: string }>((resolve, reject) => {
+        cp.exec(command, options, (error, stdout, stderr) => {
+            if (error) {
+                reject({ error, stdout, stderr });
+            }
+            resolve({ stdout, stderr });
+        });
+    });
+}
+
+let _channel: vscode.OutputChannel;
+function getOutputChannel(): vscode.OutputChannel {
+    // Creates a new entry in the "OUTPUT" panel at the bottom of the IDE.
+    if (!_channel) {
+        _channel = vscode.window.createOutputChannel('Architecture-as-code');
+    }
+    return _channel;
+}
+
+interface AacTaskDefinition extends vscode.TaskDefinition {
+    /**
+     * The task name
+     */
+    task: string;
```

Comment:
Is there a reason we can't call it `name`?

Reply:
Nope. I like name, changing it.